akka-persistence-typed/src/test/scala/docs/akka/persistence/typed/AccountExampleOptionStateWithFactory.scala

```diff
@@ -0,0 +1,93 @@
+package docs.akka.persistence.typed
+
+import akka.actor.typed.Behavior
+import akka.persistence.typed.scaladsl.PersistentBehaviors.CommandHandler
+import akka.persistence.typed.scaladsl.PersistentBehaviors6.{HandlerFactory, HandlerFactoryOption}
+import akka.persistence.typed.scaladsl.{Effect, PersistentBehaviors6}
+import docs.akka.persistence.typed.AccountExampleOptionStateWithFactory.AccountEntity.{Account, AccountCommand, AccountEvent}
+
+object AccountExampleOptionStateWithFactory {
+
+  object AccountEntity extends HandlerFactoryOption[AccountCommand,  AccountEvent, Account]{
```

Comment:
This example uses a HanlderFactory optimised for Option. 

Leakage of Option type happens in the same place as AccountExampleOption, but we have the facility of partial functions.